handler/handler.go

```diff
@@ -50,7 +64,26 @@ func (q *NatsQueue) Queue(req *queue.Request) error {
 		log.Println(err)
 	}
 
-	err = q.nc.Publish(q.Topic, out)
+	err = q.stanConn.Publish(q.Topic, out)
 
 	return err
 }
+
+func (q *NatsQueue) reconnectClient(clientID, clusterID, natsURL string) nats.ConnHandler {
+	return func(c *nats.Conn) {
+		oldConn := q.stanConn
+
+		defer oldConn.Close()
```

Comment:
With the way you handle reconnect (at the NATS low level), it is possible that the Streaming server still has the connection registered (remember, streaming server connects to NATS and clients connect to NATS, so no direct connection between client and server) because it did not detect the miss of HBs from client.

So this is pretty broken actually, because it is possible that your Connect() below will fail due to a duplicate (same clientID will be detected as duplicate by Streaming server, which will contact the old connection to its INBOX which will be valid since it is reconnected to NATS) and still you will close the old connection.

So with this current model, I would first close the old connection and then create the new one.